client/core/core.go

```diff
@@ -5288,114 +5396,142 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
 	c.sentCommits[prefix.Commit] = commitSig
 	c.sentCommitsMtx.Unlock()
 
+	// Prepare order meta data.
+	dbOrder := &db.MetaOrder{
+		MetaData: &db.OrderMetaData{
+			Host:               dc.acct.host,
+			EpochDur:           mktConf.EpochLen, // epochIndex := result.ServerTime / mktConf.EpochLen
+			FromSwapConf:       assetConfigs.fromAsset.SwapConf,
+			ToSwapConf:         assetConfigs.toAsset.SwapConf,
+			MaxFeeRate:         assetConfigs.fromAsset.MaxFeeRate,
+			RedeemMaxFeeRate:   assetConfigs.toAsset.MaxFeeRate,
+			FromVersion:        assetConfigs.fromAsset.Version,
+			ToVersion:          assetConfigs.toAsset.Version, // and we're done with the server's asset configs.
+			Options:            form.Options,
+			RedemptionReserves: redemptionReserves,
+			RefundReserves:     refundReserves,
+			ChangeCoin:         changeID,
+			ProgramID:          form.Program,
+		},
+		Order: ord,
+	}
+
+	tradeRequest := &tradeRequest{
+		mktID:        mktID,
+		route:        route,
+		dc:           dc,
+		form:         form,
+		dbOrder:      dbOrder,
+		msgOrder:     msgOrder,
+		recoveryCoin: recoveryCoin,
+		coins:        coins,
+		wallets:      wallets,
+		errCloser:    errCloser.Copy(),
+		preImg:       preImg,
+	}
+
+	errCloser.Success()
+
+	return tradeRequest, nil
+}
+
+// sendTradeRequest sends an order, processes the result, then prepares and
+// stores the trackedTrade.
+func (c *Core) sendTradeRequest(tr *tradeRequest) (*Order, error) {
+	defer tr.errCloser.Done(c.log)
```

Comment:
This diff becomes quite a bit tamer with this at the start of `sendTradeRequest`:

```go
dc, dbOrder, wallets, form := tr.dc, tr.dbOrder, tr.wallets, tr.form
preImg, recoveryCoin := tr.preImg, tr.recoveryCoin
```